recipe/install_nvcc.sh

```diff
@@ -102,7 +102,12 @@ then
   LIBCUDA_SO_CONDA_NVCC_BACKUP="\${CONDA_BUILD_SYSROOT}/lib/libcuda.so-conda-nvcc-backup"
   mv "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so" "\${LIBCUDA_SO_CONDA_NVCC_BACKUP}"
 fi
-ln -s "\${CUDA_HOME}/lib64/stubs/libcuda.so" "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so"
+if [[ "${CONDA_BUILD}" == 1 ]]
+then
+  ln -s "\${CUDA_HOME}/lib64/stubs/libcuda.so" "\${CONDA_BUILD_SYSROOT}/lib/libcuda.so"
+else
+  ln -s "\${CUDA_HOME}/lib64/stubs/libcuda.so" "\${CONDA_PREFIX}/lib/libcuda.so"
+fi
```

Comment:
It would be nice to handle this outside of conda-build as well.

We could use the output of `gcc -print-sysroot` in order to reliably get the sysroot.

Reply:
We are just exploring options here at this point.

That said, some folks are running into build issues with CMake locally and this one attempt to address that. It's possible we need to do something different in the end

Reply:
> That is a different location compared to CONDA_BUILD_SYSROOT, but it is searched by CMake

Where is `CONDA_BUILD_SYSROOT` during a conda build? I figured it would be in the `build` environment at a similar place to where `gcc -print-sysroot` points to.

> Would ${CONDA_PREFIX}/lib/stubs be safe?

I assume so. It would be nice to ship all of the cuda stub libraries in a package in addition to the driver stub.

Reply:
The locations outside of a conda build are:

* `CONDA_BUILD_SYSROOT` => `<env>/x86_64-conda-linux-gnu/sysroot`
* `gcc -print-sysroot` => `<env>/x86_64-conda-linux-gnu/sysroot`
* `CMAKE_COMPILER_SYSROOT` => `<env>/x86_64-conda-linux-gnu/sysroot/usr/`

We would want to place the stub either in `${CONDA_PREFIX}/lib/stubs` or in `$(gcc -print-sysroot)/usr` for it to be found by CMake.

Reply:
Could it go into `${gcc -print-sysroot)/usr/lib`? That seems to make sense for me and mirrors where the driver is in a normal system install.

Reply:
So then I think we can generally use `$(gcc -print-sysroot)/usr/lib` for this and ignore the `CONDA_BUILD_SYSROOT` environment variable entirely?

Reply:
Can we safely presume that `gcc` points to a conda gcc?

Reply:
The way the current recipe is, I guess not 🙃.

Do we want to allow using this package with system compilers outside of the conda environment?

Reply:
How about we use `$(CONDA_BUILD_SYSROOT)/usr/lib` since that should resolve to the same location as `$(gcc -print-sysroot)/usr/lib`

Reply:
`CONDA_BUILD_SYSROOT` is undefined within my environment so I don't think we can rely on that either.

Reply:
So our only option would be ${CONDA_PREFIX}/lib/stubs

Reply:
Technically the package depends on the `sysroot_linux-64 2.17` package (https://github.com/conda-forge/nvcc-feedstock/blob/main/recipe/meta.yaml#L57) so a sysroot would be available in any conda environment with this package.

That being said, in theory someone could be using this package with host compilers (though I don't think we really support doing so), in which case it probably wouldn't search in the conda env sysroot, but would use the system sysroot which would presumably have the driver anyway.

Reply:
Alternatively, NVIDIA upstreaming things properly based on https://github.com/conda-forge/cudatoolkit-feedstock/issues/62 would make all of this work a moot point.

Also alternatively, it's not declared in the EULA, but if redistributing the stub libraries was allowed then we could just add all of the stub libraries to the `cudatoolkit` package which would include the `libcuda.so` stub and make this a moot point as well.

Reply:
Moved to `stubs`